internal/service/projectapikey/resource_project_api_key.go

```diff
@@ -442,31 +448,32 @@ func getStateProjectAssignmentAPIKeys(d *schema.ResourceData) (newAssignments, c
 	return
 }
 
-func getAPIProjectAssignments(ctx context.Context, conn *matlas.Client, apiKeyOrgList *matlas.Root, apiKeyID string) ([]APIProjectAssignmentKeyInput, error) {
+func getAPIProjectAssignments(ctx context.Context, connV2 *admin.APIClient, apiKeyOrgList *admin.SystemStatus, apiKeyID string) ([]APIProjectAssignmentKeyInput, error) {
 	projectAssignments := []APIProjectAssignmentKeyInput{}
-	for idx, role := range apiKeyOrgList.APIKey.Roles {
-		if strings.HasPrefix(role.RoleName, "ORG_") {
-			orgKeys, _, err := conn.APIKeys.List(ctx, apiKeyOrgList.APIKey.Roles[idx].OrgID, nil)
-			if err != nil {
-				return nil, fmt.Errorf("error getting api key information: %s", err)
-			}
-			for _, val := range orgKeys {
-				if val.ID == apiKeyID {
-					for _, r := range val.Roles {
-						temp := new(APIProjectAssignmentKeyInput)
-						if strings.HasPrefix(r.RoleName, "GROUP_") {
-							temp.ProjectID = r.GroupID
-							for _, l := range val.Roles {
-								if l.GroupID == temp.ProjectID {
-									temp.RoleNames = append(temp.RoleNames, l.RoleName)
-								}
+	for idx, role := range *apiKeyOrgList.ApiKey.Roles {
```

Comment:
maybe return empty if apiKeyOrgList == nil?
And use GetRoles?

Reply:
When is passed to the method, a check has already been made that the API call has been successful, so it should always be different than nil at this point. Changed to use GetRoles()